### PR TITLE
NF: Gestures: Add Key/Card Side Selection screens & preferences serde

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description
Hopefully the last PR before the functional change, aims to reduce the amount of code to review when 

* Add `CardSideSelectionDialog` (unused)
* Add `KeySelectionDialogBuilder` (unused)
* Convert `Binding` to Kotlin
* Extract `MappableBinding` and `CardSide`
  * MappableBinding is the item which is serialised into the preferences

image
image

## Fixes
Related: #6052, some code from #8078

## How Has This Been Tested?

Unit tested and on my Android 11

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/reviewer/Binding.kt [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+class Binding private constructor(private val mModifierKeys: ModifierKeys?, private val mKeycode: Int?, private val mUnicodeCharacter: Char?, private val mGesture: Gesture?) {
+    fun toDisplayString(context: Context?): String {
+        val string = StringBuilder()
+        when {
+            mKeycode != null -> {
+                string.append(KEY_PREFIX)
+                string.append(' ')
+                string.append(Objects.requireNonNull(mModifierKeys).toString())
+                val keyCodeString = KeyEvent.keyCodeToString(mKeycode)
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/reviewer/Binding.kt:36]: https://developer.android.com/reference/android/view/KeyEvent#keyCodeToString(int) - this does not handle localisation, and I can't think of a clean way to do this.

File: AnkiDroid/src/main/res/values/10-preferences.xml [hunk 2]
```diff
@@ -111,7 +111,7 @@
     <string name="sync_fetch_missing_media_summ">Automatically fetch missing media when syncing</string>
     <string name="sync_account" maxLength="41">AnkiWeb account</string>
     <string name="sync_account_summ_logged_out">Not logged in</string>
-    <string name="sync_account_summ_logged_in">%s</string>
+    <string name="sync_account_summ_logged_in" comment="The summary for the account setting when logged in. Parameter is the username (email)">%s</string>
```
Comment (mikehardy) [inline AnkiDroid/src/main/res/values/10-preferences.xml:114]: A drive-by fix? :-)

**Review by mikehardy — APPROVED** [review]
That's a lot, but all seems fine. The key code string thing seems to be that we are sort of abusing the API, as it is documented to be for test/debug only. In that sense when the user is setting it up they are sort of testing it so it makes sense to want to show them but I also could not imagine a way to internationalize them. The illegal unicode thing returning unknown was curious to me, could maybe use an explanatory comment. In general though I know this is one part of a mountain of work and I've no intention of getting in the way of it since you've definitely adopted the subsystem here - merge when you're happy IMHO

**Comment by mikehardy** [comment]
Would be nice to see a rebase - this shouldn't be happening anymore?

```
com.ichi2.anki.DeckPickerTest > checkIfClickOnCountsLayoutOpensStudyOptionsOnMobile[TestingAVD(AVD) - 11] FAILED 
	androidx.test.espresso.base.RootViewPicker$RootViewWithoutFocusException: Waited for the root of the view hierarchy to have window focus and not request layout for 10 seconds. If you specified a non default root matcher, it may be picking a root that never takes focus. Root:
```

File: AnkiDroid/src/main/res/values/10-preferences.xml (hunk 2 above)
Comment (david-allison-1) [reply AnkiDroid/src/main/res/values/10-preferences.xml:114]: No, I've introduced another constant which is also `%s`, so this is needed to remove the lint error of duplicate strings

Comment (mikehardy) [reply AnkiDroid/src/main/res/values/10-preferences.xml:114]: Ah ha! Was a good change either way
